src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -281,7 +283,16 @@ private static IEnumerable<FunctionOverload> GetSystemOverloads(IFeatureProvider
 
             yield return new FunctionOverloadBuilder("first")
                 // TODO even with non-literal types, some type arithmetic could be performed
-                .WithReturnResultBuilder(TryDeriveLiteralReturnType("first", LanguageConstants.Any), LanguageConstants.Any)
+                .WithReturnResultBuilder((binder, fileResolver, diagnostics, arguments, argumentTypes) =>
+                {
+                    return argumentTypes[0] switch
+                    {
+                        TypedArrayType typedArrayType => new(typedArrayType.Item.Type),
+                        TupleType tupleType when tupleType.Items.Length == 0 => new(LanguageConstants.Null),
```

Comment:
There are two special cases that need to be handled with tuples. `null` is the correct type if the tuple is empty, but otherwise the return type should be `tupleType.Items[0].Type`. For example, a tuple type like `[string, int]` should create a return value of `string` for the `first` function.

Both cases can be handled by something like `tupleType.Items.FirstOrDefault()?.Type ?? LanguageConstants.Null`